cli/flox/src/commands/init/go.rs

```diff
@@ -0,0 +1,479 @@
+
+use super::{
+    format_customization,
+    get_default_package_if_compatible,
+    try_find_compatible_version,
+    InitHook,
+    ProvidedPackage,
+    ProvidedVersion,
+    AUTO_SETUP_HINT,
+};
+use crate::utils::dialog::{Dialog, Select};
+use crate::utils::message;
+
+const GO_MOD_FILENAME: &str = "go.mod";
+const GO_WORK_FILENAME: &str = "go.work";
+
+const GO_HOOK: &str = indoc! {"
+    # Install Go depedencies
+    go get ."
+};
+
+/// The Go hook handles installation and configuration suggestions for projects using Go.
+/// The general flow of the Go hook is:
+///
+/// - [Self::new]: Detects [GoModuleSystem] files in the current working directory.
+/// - [Self::should_run]: Returns whether a valid module system was detected
+///   in the current working directory, i.e. `false` if the [Self::module_system]
+///   is [GoModuleSystemKind::None], else returns `true`.
+/// - [Self::prompt_user]: Prints the customization from [Self::get_init_customization]
+///   if user commands to do so. Else, return true or false based on whether
+///   the user wants the customization.
+/// - [Self::get_init_customization]: Returns a customization based on [Self::module_system].
+pub(super) struct Go {
+    /// Stores the customization values required to generate a customization with
+    /// [Self::get_init_customization].
+    /// Is initialized in [Self::new].
+    module_system: GoModuleSystemKind,
+}
+
+impl Go {
+    /// Creates and returns the Go hook with the detected module system.
+    pub fn new(path: &Path, flox: &Flox) -> Result<Self> {
+        let module_system = Self::detect_module_system(path, flox)?;
+
+        Ok(Self { module_system })
+    }
+
+    /// Determines which [GoModuleSystemKind] is being used.
+    /// Since the [GO_WORK_FILENAME] file declares a multiple module based workspace, it takes
+    /// precedence over any other [GO_MOD_FILENAME] file that could possibly be found.
+    fn detect_module_system(path: &Path, flox: &Flox) -> Result<GoModuleSystemKind> {
+        if let Some(go_work) = GoWorkspaceSystem::try_new_from_path(path, flox)? {
+            return Ok(GoModuleSystemKind::Workspace(go_work));
+        }
+
+        if let Some(go_mod) = GoModuleSystem::try_new_from_path(path, flox)? {
+            return Ok(GoModuleSystemKind::Module(go_mod));
+        }
+
+        Ok(GoModuleSystemKind::None)
+    }
+}
+
+impl InitHook for Go {
+    /// Returns `true` if any valid module system file was found.
+    ///
+    /// [Self::prompt_user] and [Self::get_init_customization]
+    /// are expected to be called only if this method returns `true`!
+    fn should_run(&mut self, _path: &Path) -> Result<bool> {
+        Ok(self.module_system != GoModuleSystemKind::None)
+    }
+
+    /// Returns `true` if the user accepts the prompt. In that case,
+    /// the hook customizes the manifest with the default Go environment.
+    fn prompt_user(&mut self, _path: &Path, _flox: &Flox) -> Result<bool> {
```

Comment:
quib (NB!): I think we should really break this out into a function or out of default impl this in #1266, but no need to worry about this here!